mempool/cat/cache.go

```diff
@@ -0,0 +1,291 @@
+package cat
+
+import (
+	"container/list"
+	"time"
+
+	tmsync "github.com/tendermint/tendermint/libs/sync"
+	"github.com/tendermint/tendermint/types"
+)
+
+// LRUTxCache maintains a thread-safe LRU cache of raw transactions. The cache
+// only stores the hash of the raw transaction.
+// NOTE: This has been copied from mempool/cache with the main diffence of using
+// tx keys instead of raw transactions.
+type LRUTxCache struct {
+	staticSize int
+
+	mtx      tmsync.Mutex
+	cacheMap map[types.TxKey]*list.Element
+	list     *list.List
```

Comment:
I was wondering why we are using both `cacheMap` and the `list`? It seems they maintain the same exact data and undertake the same modifications in all the methods, so it seems a bit redundant. Can you shed some light on this for me?